lbz/authz.py

```diff
@@ -22,24 +22,27 @@
 
 
 class Authorizer:
-    def __init__(self, auth_jwt: str, resource_name: str, permission_name: str):
+    def __init__(self, auth_jwt: str, resource_name: str, permission_name: str, auth_scope: dict = None):
         self.outcome = DENY
         self.allowed_resource = None
         self.denied_resource = None
         self.resource = resource_name
         self.permission = permission_name
         self.allow = {}
         self.deny = {}
-        self._set_policy(auth_jwt)
+        self._set_policy(auth_jwt, auth_scope)
 
     def __repr__(self):
         return (
             f"Authorizer(auth_jwt=<jwt>, resource_name='{self.resource}', "
             f"permission_name='{self.permission}')"
         )
 
-    def _set_policy(self, auth_jwt: str):
-        policy = decode_jwt(auth_jwt)
+    def _set_policy(self, auth_jwt: str, auth_scope: dict = None):
+        if auth_scope:
```

Comment:
I'd do `policy = auth_scope if auth_scope is not None else decode_jwt(auth_jwt)` (not sure what exactly should happen if `auth_scope == {}`)